rclcpp/src/rclcpp/executors/executor_entities_collection.cpp

```diff
@@ -43,10 +43,10 @@ size_t ExecutorEntitiesCollection::remove_expired_entities()
 {
   auto remove_entities = [](auto & collection) -> size_t {
       size_t removed = 0;
-      for (auto it = collection.begin(); it != collection.end(); ) {
+      for (auto it = collection.begin_ordered(); it != collection.end_ordered(); ) {
         if (it->second.entity.expired()) {
           ++removed;
-          it = collection.erase(it);
+          it = collection.erase_ordered(it);
```

Comment:
This change was not strictly necessary, but I reasoned it was slightly faster to provide the iterator for erasure to the vector and look up the iterator for the map, rather than the other way around, which is what `erase_ordered` does.